### PR TITLE
Make image config param optional

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
In light of containerless tasks it no longer makes sense to have image be a required configuration parameter. Rather, PythonAutoContainer tasks should validate that the image config is provided at serialization

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1031

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/core/context_manager.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def look_up_image_info(name: str, tag: str, optional_tag: bool = False) -> Image
 
 def get_image_config(img_name: str = None) -> ImageConfig:
```
Comment (cosmicBboy) [inline flytekit/core/context_manager.py:80]: nit: `Optional[str]`

**Review by cosmicBboy — APPROVED** [review]

**Review by wild-endeavor — APPROVED** [review]
yessss

File: flytekit/core/context_manager.py (hunk 1 above)
Comment (katrogan) [reply flytekit/core/context_manager.py]: done, thanks

**Review by cosmicBboy — APPROVED** [review]
